crates/oxc_semantic/src/symbol/mod.rs

```diff
@@ -0,0 +1,47 @@
+//! Symbol and Symbol Table for tracking of semantics of variables
+
+pub mod id;
+pub mod table;
+
+use oxc_ast::{Atom, Span};
+
+pub use self::{id::SymbolId, table::SymbolTable};
+
+#[derive(Debug)]
+pub struct Symbol {
+    id: SymbolId,
+    name: Atom,
+    span: Span,
+}
```

Comment:
The symbol here will grow so big (> 128 bytes) if we look at TypeScript's symbol. I wonder if we should change this Structure of arrays (SoA)?